x-pack/solutions/security/packages/navigation/src/navigation_tree/investigations_navigation_tree.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { NodeDefinition } from '@kbn/core-chrome-browser';
+import { SecurityPageName, SecurityGroupName } from '../constants';
+import { SecurityLinkGroup } from '../link_groups';
+import { securityLink } from '../links';
+
+export const createInvestigationsNavigationTree = (): NodeDefinition => ({
+  id: SecurityGroupName.investigations,
+  title: SecurityLinkGroup[SecurityGroupName.investigations].title,
+  renderAs: 'panelOpener',
+  children: [
+    {
+      id: SecurityPageName.timelines,
+      link: securityLink(SecurityPageName.timelines),
+      renderAs: 'item',
+      children: [
+        {
+          id: SecurityPageName.timelinesTemplates,
+          link: securityLink(SecurityPageName.timelinesTemplates),
+        },
+      ],
+    },
+    {
+      id: SecurityPageName.notes,
+      link: securityLink(SecurityPageName.notes),
+    },
+    {
+      link: 'osquery',
```

Comment:
does `osquery` not also need an id? Is this all because it's technically not under `security/`?

Reply:
We only need the `id` to be defined for Security links. It should be a `SecurityPageName` or a `SecurityGroupName`. We use it for registering the SecuritySolution plugin deepLinks:

https://github.com/semd/kibana/blob/f9e33654bd5920ebd79539639a08092a2f3118f8/x-pack/solutions/security/plugins/security_solution/public/app/links/deep_links.ts#L81-L106

The rest of (external) links don't need the `id` property to exist. It can be defined if we need to reference the link for some reason (like `id: 'stack_management'`), but otherwise we can omit it. Only the `link` property is required.

I will update the docs with this information when this is merged.